### PR TITLE
fix: Fix get promotions page and add loading icon

### PR DESCRIPTION
**Pull request by ericrkuo**
What's in this PR:
- If we called GET /promotions with the userId, our request would fail if we added any additional query parameters, for e.g. if we add additional filters like cuisine type or day of week. 
   - I think there was something strange going on with how we were constructing the URL
   - I did some refactoring and used axios's ability to specify request params
- Added loading icon when we add filters, use full text search, when refresh page and promotions are loading, etc

GetPromoSpinner

### CODE REVIEW

File: src/components/PromotionList.tsx
Comment (andrewzulaybar) [inline src/components/PromotionList.tsx]: Thanks for adding this!!! I'm wondering if we can use the same spinner that we're using right now in App.less, I think we just have to pass in an indicator icon

File: src/services/PromotionService.ts
Comment (andrewzulaybar) [inline src/services/PromotionService.ts:35]: AH YES OF COURSE this is a lot better 😅

File: src/services/PromotionService.ts
Comment (andrewzulaybar) [inline src/services/PromotionService.ts:24]: We can remove the unused import above!

File: src/services/PromotionService.ts
Comment (andrewzulaybar) [inline src/services/PromotionService.ts]: We can remove this too 😛

**Review by andrewzulaybar — APPROVED** [review]
Thanks for doing this and cleaning things up -- I like it a lot better like this! Just left a few comments, but should be good to go after addressing them

**Comment by andrewzulaybar** [comment]
Oh also let me know if you find other things that seem off! Especially styling I can help out with that :)

File: src/components/PromotionList.tsx
Comment (ericrkuo) [reply src/components/PromotionList.tsx]: Ohhh cool, I'll take a look at it!

File: src/services/PromotionService.ts
Comment (ericrkuo) [reply src/services/PromotionService.ts:24]: Ohya good point! We'll need to add it back once we merge in the BE sort PR

File: src/components/PromotionList.tsx
Comment (ericrkuo) [reply src/components/PromotionList.tsx]: Good call! This icon looks prettier now!

Comment (ericrkuo) [reply src/components/PromotionList.tsx]: GetPromoSpinner2
This is what it looks like now!

**Comment by ericrkuo** [comment]
> Oh also let me know if you find other things that seem off! Especially styling I can help out with that :)

@andrewzulaybar for sure! I'll be continuing to look for any bugs or areas we can improve

**Pull request merged**
